src/main/java/org/generationcp/middleware/util/serializer/SynonymPropertySerializer.java

```diff
@@ -0,0 +1,33 @@
+package org.generationcp.middleware.util.serializer;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import org.apache.commons.collections.CollectionUtils;
+import org.generationcp.middleware.api.brapi.v2.germplasm.Synonym;
+import org.generationcp.middleware.service.api.BrapiView;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SynonymPropertySerializer extends JsonSerializer<List<Synonym>> {
+
+	@Override
+	public void serialize(final List<Synonym> value, final JsonGenerator gen, final SerializerProvider serializers)
+		throws IOException {
+
+		if (serializers.getActiveView() != null && CollectionUtils.isNotEmpty(value)) {
+			if (BrapiView.BrapiV1_3.class.equals(serializers.getActiveView()) || BrapiView.BrapiV1_2.class
+				.equals(serializers.getActiveView())) {
+				gen.writeObject(value.stream().map(Synonym::getSynonym).collect(Collectors.toList()));
+			} else {
+				gen.writeObject(value);
+			}
+		} else {
```

Comment:
Why do we need this final else clause here (ie. when there are no synonyms)?

Reply:
If synonym is empty, we need to write/return it as is, if not, JsonSerializer will return an error "Can not write a field name, expecting a value"